### ISSUE TITLE
💅 applying fixes does not support formatter singleQuote configuration

### ISSUE THREAD

**Issue by paularmstrong** [biomejs/biome#1903]
### Environment information

```bash
CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "screen-256color"
  JS_RUNTIME_VERSION:           "v20.11.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/3.3.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### Rule name

lint/style/noUnusedTemplateLiteral

### Playground link

playground does not include rule

### Expected result

```json
{
  "javascript": {
    "formatter": {
      "quoteStyle": "single"
    }
  }
}
```

input:
```js
const foo = `foobar`;
```

command:
```sh
biome lint --apply-unsafe .
```

expected output
```js
const foo = 'foobar';
```

actual output:
```js
const foo = "foobar";
```

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
